code/modules/mob/living/living_defense.dm

```diff
@@ -167,6 +167,10 @@
 			blocked = TRUE
 		else
 			playsound(loc, 'sound/weapons/genhit.ogg', 50, TRUE, -1) //Item sounds are handled in the item itself
+			if(isvendor(AM) == FALSE) //Vendors have special interactions!
```

Comment:
```suggestion
			if(!isvendor(AM) && !iscarbon(AM)) //Vendors have special interactions, while carbon mobs already generate visible messages!
```

Reply:
Oh right, i forgot about other types of non carbon mobs!